src/implementation/Client/GRPCClient/binding.ts

```diff
@@ -12,8 +12,47 @@ limitations under the License.
 */
 
 import GRPCClient from './GRPCClient';
-import { InvokeBindingRequest, InvokeBindingResponse } from '../../../proto/dapr/proto/runtime/v1/dapr_pb';
+import {
+  InvokeBindingRequest as AutogeneratedInvokeBindingRequest,
+  InvokeBindingResponse as AutogeneratedInvokeBindingResponse,
+} from '../../../proto/dapr/proto/runtime/v1/dapr_pb';
 import IClientBinding from '../../../interfaces/Client/IClientBinding';
+import { types } from 'util';
+import { InvokeBindingResponse } from '../../../types/binding/InvokeBindingResponse.type';
+
+export class BindingRequest {
```

Comment:
Do we need to export `BindingRequest` and `BindingResponse`?

Can we use [parameter properties](https://www.typescriptlang.org/docs/handbook/2/classes.html#parameter-properties) instead of members? Class definition will be cleaner.

Reply:
Well, do we need these classes at all? `send` function can directly call `makeRequest` (instead of going via `invoke`, and `BindingRequest` or `BindingResponse` won't be needed.

Reply:
The reason for this is that we do not want to make destructive changes to the `send` function. The long-term plan is to add `invoke(req: BindingRequest): Promise<BindingResponse>` and `invoke(name: string, operation: string, data: any, metadata: Record<string, string>):   Promise<BindingResponse>`

Reply:
`invoke` isn't a part of the public API, right? `IClientBinding` only has a `send`. Can you please share why do we need to add `invoke` and its two overloads?

Reply:
Are you talking about changing the API name from `client.binding.send` to `client.binding.invoke`? https://docs.dapr.io/reference/api/bindings_api/#invoking-output-bindings we technically need only one method in our `IClientBinding`, which we have as `send` today. 

Do we really need the `BindingRequest` class? Isn't it simpler to use the params `name`, `operation`, `data`, and `metadata` directly? It makes sense from a consumer point of view and eliminates the need to pack and unpack the data. The class `BindingRequest` itself provides no value (it's more like a struct).

Similar argument for `BindingResponse`. Since we already have `InvokeBindingResponse`, do we need to introduce yet another class here?

/cc @XavierGeerinck

Reply:
Agreed, deprecating `send` in favor of `invoke` is not something we want to do at this moment.

Regarding classes, I personally also would refrain from adding extra abstraction classes. The gRPC Classes build from dapr/dapr already provide a DTO, this would just add another DTO on top.

I can however see where this is coming from as the Java SDK does it through this way.